src/components/PokeList.js

```diff
@@ -11,53 +12,129 @@ const PokeList = ({
   allPokemons,
 }) => {
   const { id, name, sprites, types } = fetchedData;
-  let typeTags, sprite;
+  const [selectTypeOption, setSelectTypeOption] = useState("");
+  const [selectAbilityOption, setSelectAbilityOption] = useState("");
+  // abilityOptions: array of sorted ability names for drop-down
+  const [abilityOptions, setAbilityOptions] = useState([]);
+  // state for sorting toggle
+  const [sortType, setSortType] = useState(SortTypes.DEX_NO);
+  // stores filtered Pokemon array returned by sortAndFilter()
+  const [filteredPokemons, setFilteredPokemons] = useState([]);
+  const [userDidSort, setUserDidSort]= useState(false)
+
+  const setSortTypeFlags = (sortType)=>{
+    setSortType(sortType);
+    setUserDidSort(true);
+  }
 
+  // generate tags, check for tags and types
+  let typeTags, sprite;
   if (types && sprites) {
     sprite = sprites["other"]["official-artwork"]["front_default"];
     typeTags = types.map((obj, i) => {
       return <span key={i}>{obj["type"]["name"]} </span>;
     });
   }
-  // <PokeCard dexNo={id} name={name} sprites={sprites} typeTags={typeTags}/>
+  // array of Pokemon objects sorted by keys: name, weight, height, dex no
+  const sortByName = (pokemonList) => {
+    return pokemonList.sort((a, b) => {
+      const nameA = a["name"];
+      const nameB = b["name"];
+      if (nameA < nameB) return -1;
+      if (nameA > nameB) return 1;
+      return 0;
+    });
+  };
+
+  const sortByWeight = (pokemonList) => {
+    return pokemonList.sort((a, b) => {
+      const weightA = a["weight"];
+      const weightB = b["weight"];
+      return weightA - weightB;
+    });
+  };
+
+  const sortByHeight = (pokemonList) => {
+    return pokemonList.sort((a, b) => {
+      const heightA = a["height"];
+      const heightB = b["height"];
+      return heightA - heightB;
+    });
+  };
+  const sortByDexNumber = (pokemonList) => {
+    return pokemonList.sort((a, b) => {
+      const id1 = a["id"];
+      const id2 = b["id"];
+      return id1 - id2;
+    });
+  };
```

Comment:
should we stick all these sort functions in their own js file during the refactor stage?